src/filters.rs

```diff
@@ -506,65 +508,103 @@ pub struct BboxFilter<'a> {
     kb: KeyBuilder,
     bbox: Vec<u8>,
     term_ordinal: Option<usize>,
+    last_returned: Option<(u64, Vec<u64>)>, // (seq, arraypath)
 }
 
 impl<'a> BboxFilter<'a> {
     pub fn new(snapshot: Rc<Snapshot<'a>>, kb: KeyBuilder, bbox: [f64; 4]) -> BboxFilter<'a> {
         let mut bbox_vec = Vec::with_capacity(32);
-        bbox_vec.extend_from_slice(&bbox[0].to_le_bytes());
-        bbox_vec.extend_from_slice(&bbox[2].to_le_bytes());
-        bbox_vec.extend_from_slice(&bbox[1].to_le_bytes());
-        bbox_vec.extend_from_slice(&bbox[3].to_le_bytes());
+        // RTree expects bbox in order: [x_min, x_max, y_min, y_max]
+        // Input is [west, south, east, north] = [x_min, y_min, x_max, y_max]
+        bbox_vec.extend_from_slice(&bbox[0].to_le_bytes()); // x_min (west)
+        bbox_vec.extend_from_slice(&bbox[2].to_le_bytes()); // x_max (east)
+        bbox_vec.extend_from_slice(&bbox[1].to_le_bytes()); // y_min (south)
+        bbox_vec.extend_from_slice(&bbox[3].to_le_bytes()); // y_max (north)
 
         BboxFilter {
             snapshot,
             iter: None,
             kb,
             bbox: bbox_vec,
             term_ordinal: None,
+            last_returned: None,
         }
     }
 
     /// Function to deserialize the Arraypaths
     fn from_u8_slice(slice: &[u8]) -> Vec<u64> {
-        let u64_slice =
-            unsafe { std::slice::from_raw_parts(slice.as_ptr() as *const u64, slice.len() / 8) };
-        u64_slice.to_vec()
+        slice
+            .chunks_exact(8)
+            .map(|chunk| {
+                u64::from_le_bytes(
+                    chunk
+                        .try_into()
+                        .expect("array path entries are multiples of 8 bytes"),
+                )
+            })
+            .collect()
     }
 }
 
 impl<'a> QueryRuntimeFilter for BboxFilter<'a> {
     fn first_result(&mut self, start: &DocResult) -> Option<DocResult> {
-        let query = self
-            .kb
-            .rtree_query_key(start.seq, std::u64::MAX, &self.bbox);
+        // For array queries, we need to consider the context
+        let min_seq = if self.kb.arraypath.is_empty() {
+            // Not an array query, use normal start seq
+            start.seq
+        } else if start.seq > 0 && start.arraypath.len() == self.kb.arraypath.len() {
+            // Array query for a specific document - search within that document
+            // by using seq-1 to ensure we get all array elements
+            start.seq.saturating_sub(1)
+        } else {
+            start.seq
+        };
```

Comment:
Having two conditional code paths with the same value feels strange. It's about whether it's an array query or not. So I'd rather do something like (not tested):

```suggestion
        // For array queries, we need to consider the context
        let min_seq = if self.kb.arraypath > 0 && start.seq > 0 && start.arraypath.len() && self.kb.arraypath.len()` {
            // Array query for a specific document - search within that document
            // by using seq-1 to ensure we get all array elements
            start.seq.saturating_sub(1)
        } else {
            start.seq
        };
```